test/test_python_symint.py

```diff
@@ -0,0 +1,84 @@
+from typing import Iterator, List
+import torch
+from enum import Enum
+import torch
+#from base_tensor import BaseTensor
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.utils._pytree import tree_map
+import contextlib
+
+@contextlib.contextmanager
+def no_dispatch() -> Iterator[None]:
+    guard = torch._C._DisableTorchDispatch()  # type: ignore[attr-defined]
+    try:
+        yield
+    finally:
+        del guard
+
+class Op(Enum):
+    SIZE = 1
+    ADD = 2
+
+class AOTAutogradSymInt:
+    def __init__(self, op: Op, children: List["AOTAutogradSymInt"]):
+        self._op = op
+        self._children = children
+
+    def __str__(self):
+        return f"{self._op}"
+
+    def __add__(self, other):
+        return AOTAutogradSymInt(Op.ADD, [self, other])
+
+
+a = torch.rand(2, 2)
+
+sym1 = torch._C.SymbolicIntNode.new_symint(AOTAutogradSymInt(Op.SIZE, [a, 0]))
+sym2 = torch._C.SymbolicIntNode.new_symint(AOTAutogradSymInt(Op.SIZE, [a, 0]))
+
+class EmptyTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, elem):
+        symints = [torch._C.SymbolicIntNode.new_symint(AOTAutogradSymInt(Op.SIZE, [elem, i])) for i in range(len(elem.size()))]
+        return torch.Tensor._make_wrapper_subclass(
+            cls, symints,
+            strides=elem.stride(), storage_offset=elem.storage_offset(),
+            dtype=elem.dtype, layout=elem.layout, requires_grad=elem.requires_grad,
+            device=elem.device
+        )
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        def inflate(t):
+            if isinstance(t, cls):
+                with no_dispatch():
+                    return torch.ones_like(t, device=t.device)
+            else:
+                return t
+
+        def deflate(t):
+            if isinstance(t, torch.Tensor) and not isinstance(t, cls):
+                return EmptyTensor(t)
+            else:
+                return t
+
+        return tree_map(deflate, super().__torch_dispatch__(
+            func, types,
+            tree_map(inflate, args),
+            tree_map(inflate, kwargs)
+        ))
```

Comment:
inflate/deflate copy paste looks wrong

Reply:
Yes, I just needed it to let me test `t.size()`